### ISSUE TITLE
named schemas

### ISSUE THREAD

**Issue by Dgreyvenstein** [JuergenGutsch/graphql-aspnetcore#37]
Hi, 
I've just started working with the library, so I may be approaching this in the wrong way. To support multiple queries, it seems I need multiple endpoints and schemas. To allow for this, schemas need to be named. The following non-breaking change makes this possible (adding the schemaName parameter).

~~~~csharp
        public static IGraphQlBuilder AddGraphQl(this IServiceCollection services,
            Action<SchemaConfiguration> configure, string schemaName = null)
        {
            if (services == null)
                throw new ArgumentNullException(nameof(services));

            if (configure == null)
                throw new ArgumentNullException(nameof(configure));

            services.AddSingleton<IDocumentExecuter, DocumentExecuter>();

            var builder = new GraphQlBuilder(services);

  // **** use schemaName parameter here
            var schema = new SchemaConfiguration(schemaName);
            configure(schema);

            return builder.AddSchema(schema);
        }
~~~~

Then in ConfigureServices(...)
~~~~csharp
services.AddGraphQl(schema =>
{
	schema.SetQueryType<BooksQuery>();
}, "book");
~~~~

and in Configure(...)
~~~~csharp
app.UseGraphQl("/graphql/book", options =>
{
	options.SchemaName = "book"; 
	options.FormatOutput = false;
	options.ComplexityConfiguration = new ComplexityConfiguration { MaxDepth = 15 };

});
~~~~

Could you add this change?

**Comment by vertonghenb** [comment]
I'm not sure if this is really following the GraphQL spec, nice a Schema only has 1 Query and 1 Mutation Type, on the query type there are multiple fields declared like books, book(id), users, user(id),... hence multiple queries.

**Comment by Dgreyvenstein** [comment]
Thanks for looking into it.

My issue was actually running multiple schemas on different endpoints in one aspnet core app.

**Comment by JuergenGutsch** [comment]
Hi @Dgreyvenstein 
with the current pre-release on myget it is possible to provide multiple endpoints

**Comment by JuergenGutsch** [comment]
I recently released a 1.1.4 on NuGet. This version supports multiple endpoints.
